src/main/java/com/google/sps/data/NSWGovAdaptor.java

```diff
@@ -0,0 +1,163 @@
+  }
+
+  private Coordinates getCoord(Integer siteId) {
+
+    Coordinates siteCoord = new Coordinates(); 
+    try {
+
+      //Check if the siteId is in the map or not. If not, update the map first
+      if (this.dataMap.containsKey(siteId) == false ) {
+        this.updateSiteInfo();
+      }
+      
+      siteCoord.setCoordinates(this.dataMap.get(siteId).lng, this.dataMap.get(siteId).lat);
+
+    } catch (final NullPointerException e) {
+
+      //Indent continuation at least +4 from original line
+      System.out.println("getCoord getMessage(): " + e.getMessage() + 
+          "\nNull pointer exception caught, because null is returned as siteId Key is not in the hashmap" +
+          "\nFailure to get coordinates. Default coord (-1,-1 ) is expected to be returned."); 
+
+    } catch (final Exception e) {
+      System.out.println("getCoord getMessage(): " + e.getMessage() + 
+          "\nFailure to get coordinates. Default coord (-1,-1 ) is expected to be returned."); 
+    } 
+
+    //If siteId was found in map, it's locations coordinates will be returned
+    //If siteId was not found in map, or any other errors were happened, (-1, -1) will be returned
+    return siteCoord;
+  }
+
+
+  private void updateSiteInfo() throws Exception {
+    //Creating a get request
+    URL url = new URL("https://data.airquality.nsw.gov.au/api/Data/get_SiteDetails");
+    HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+    connection.setRequestMethod("GET");
+    
+    connection.setRequestProperty("Content-Type", "application/json");
+    
+    int responseCode = connection.getResponseCode();
+
+    ArrayList<GovSiteDetails> convertedlist = new ArrayList<GovSiteDetails>();
+    
+    System.out.println(java.time.LocalDate.now());  
+
+    try {
+      if (responseCode == HttpURLConnection.HTTP_OK) {
+        BufferedReader br = new BufferedReader(new InputStreamReader(connection.getInputStream()));
+        StringBuilder response = new StringBuilder();
+        String responseLine; 
+        while ((responseLine = br.readLine()) != null) {
+          response.append(responseLine.trim());
+        }
+  
+        String responseString = response.toString();
+        Gson gson = new Gson();
+        convertedlist = gson.fromJson(responseString, new TypeToken<ArrayList<GovSiteDetails>>() {}.getType());
+
+        //Convert the arrayList into a map. 
+        for(GovSiteDetails details : convertedlist){
+          dataMap.put(details.siteId , details);
+        } 
+      } else {
+
+        //Throw custom exception when the response code is not 200
+        throw new HTTPStatusCodeException("HTTP Status Code is not 200");
+      }
+    } catch (final Exception e) {
+      System.out.println("updateSiteInfo getMessage(): " + e.getMessage()); 
+    }
+  }
+
+  //Post request to extract the AQI from the government API, given a desired date
+  private ArrayList<NSWGovAQDataPoint> extractAQI(LocalDate inputDate) throws Exception {
```

Comment:
Pls share more context about having two `extractAQI ` method? How are you going to use it?

Reply:
oh nope, the two are the same. I basically merged it all NSWGovAdaptor so all the methods are in side it. I asked above if you think keeping it separate is better or not, but i'll remove it rn

Reply:
ok